File: Rust/167. Two Sum II - Input Array Is Sorted.rs
impl Solution {
    pub fn two_sum(numbers: Vec<i32>, target: i32) -> Vec<i32> {
        let mut left: usize = 0;
        let mut right: usize = numbers.len() - 1;

        while left <= right {
            match numbers[left] + numbers[right] {
                sum if sum == target => {
                    break;
                }
                sum if sum > target => {
                    right -= 1;
                }
                _ => {
                    left += 1;
                }
            }
        }

        vec![(left + 1) as i32, (right + 1) as i32]
    }
}

File: Java/167. Two Sum II - Input Array Is Sorted.java
class Solution {
    public int[] twoSum(int[] numbers, int target) {
        int l = 0, r = numbers.length - 1;
        int m;

        while (l < r) {
            m = numbers[l] + numbers[r];

            if (m == target)
                break;
            else if (m > target)
                r--;
            else
                l++;
        }

        return new int[] { l + 1, r + 1 };
    }
}